plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -443,13 +444,40 @@ class producer_plugin_impl : public std::enable_shared_from_this<producer_plugin
          schedule_production_loop();
       }
 
-      void on_incoming_transaction_async(const packed_transaction_ptr& trx, bool persist_until_expired, next_function<transaction_trace_ptr> next) {
+      void on_incoming_transaction_sync(const packed_transaction_ptr &trx, bool persist_until_expired, bool read_only, next_function<transaction_trace_ptr> next) {
```

Comment:
Doesn't look like read_only arg is used.

Reply:
Now properly used in constructing transaction metadata.